ui/dashboards/src/components/Variables/Variable.tsx

```diff
@@ -177,7 +178,8 @@ function TextVariable({ name }: TemplateVariableProps) {
   const ref = useRef<HTMLInputElement>(null);
   return (
     <TextField
-      ref={ref}
+      inputRef={ref}
+      key={(state?.value as string) ?? ''}
```

Comment:
Since we use defaultValue and ref, this makes sure the component gets the latest value prop. Alternatively, could refactor this to do something similar with useState

Reply:
Man, I wish MUI played better with uncontrolled components. We had this whole debate internally when figuring out the component library guidelines and ended up settling on "make everything controlled" because MUI's so inconsistent in its support. Would be good if we could try and refactor just for consistency everywhere. Gonna be weird if we're doing it differently in different places based on whether the MUI component we're using supports it.

Reply:
Yep agreed sounds good!